connector/MongoDBAtlasODBC.pq

```diff
@@ -51,13 +52,71 @@ GetCredentialConnectionString = () as record =>
     in
         CredentialConnectionString;
 
-// SQL-1117: Implement ProcessUri Function
-ProcessUri = (uri as text) as record => 
+NullFieldCheck = (field as any, fieldName as text ) =>
+    let 
+        Result = if field <> null then Record.AddField([], fieldName, field) else []
+    in
+        Result;
+
+ProcessAuthSrc = (queryEntry as any, pathEntry as any) =>
     let
-        Result = []
+        DefaultAuthSrc = "admin",
+        FieldName = "AUTH_SRC",
+        // The 'authSource' connection option takes precedence over the /defaultauthdb value in the connection string.
+        // If both are unspecified, then authSource defaults to 'admin'.
+        AuthSrc =    if queryEntry <> null then
+                        queryEntry
+                    else if pathEntry <> null and Text.Length(Text.Remove(pathEntry, "/")) > 0 then
+                        Text.Remove(pathEntry, "/")
+                    else
+                        DefaultAuthSrc,
+        Result = Record.AddField([], FieldName, AuthSrc)
     in 
         Result;
 
+ProcessUri = (uri as text) as any => 
+    let
+        DefaultPort = "27017",
+        UriParts = Uri.Parts(uri),
+        BadUri = UriParts[Host] = "" or UriParts[Fragment] <> ""
+            or UriParts[UserName] <> "" or UriParts[Password] <> "",
+        // http and https schemes automatically set port values to 80 and 443 respectively.
+        // The following checks whether the user has set the port values explicitly to 80 or 443.
+        // An unknown scheme will set the port value to -1 if it is not specified
+        Port =  if  (UriParts[Port] = 80 and not Text.Contains(uri, ":80")) or
+                    (UriParts[Port] = 443 and not Text.Contains(uri, ":443")) or
+                    UriParts[Port] = -1 then
+                    DefaultPort
+                else
+                    Number.ToText(UriParts[Port]),
+        SSL = NullFieldCheck(UriParts[Query][ssl]?, "SSL"),

```

Comment:
tls and ssl are the same option based on the doc. I see that you pass them in the array, I don't really know why we need to do that. Is there a field for SSL/TLS too? If not, can we let the driver take care of this? 
The [ssl](https://www.mongodb.com/docs/manual/reference/connection-string/#mongodb-urioption-urioption.ssl) option is equivalent to the [tls](https://www.mongodb.com/docs/manual/reference/connection-string/#mongodb-urioption-urioption.tls) option.

ODBC does not support Properties the way JDBC does.
SQLDriverConnect and SQLBrowseConnect use a connection string. SQLConnect needs a DSN.
If we can simply have something which matches the API, it would be much more easier to handle.
I understand that we might want add a field for the DB because it is more straightforward this way, so the gymnastic with authSource could still be required. Although, I'm not even sure since the DB passed here is used to setDatabase on the Rust client after establishing the connection.